homeassistant/components/modbus/__init__.py

```diff
@@ -95,6 +94,10 @@ def setup(hass, config):
     for client_config in config[DOMAIN]:
         hub_collect[client_config[CONF_NAME]] = ModbusHub(client_config)
 
+        # Set first hub in a list as a default
```

Comment:
@janiversen please check this code. I just found a bug, which was in the Modbus code for a very long time. Each entity has an optional parameter `hub`. If it's not specified, a `DEFAULT_HUB` value gets assigned instead.

When such an entity wants to access a "default" hub, it looks into Modbus hub dictionary and asks for a "default" hub, which doesn't exist. This small change fixes that as I'm assigning the first defined hub in a list as a default, i.e., two keys point to one `ModbusHub` instance.

In case you'll think of a better way how to handle this situation, please let me know. One option would be to get rid of the `default hub` completely, and make it a required parameter.

Reply:
My preference is to drop “default hub” and make hub required. this is not a real breaking change since default hub dis not work (good work catching that bug<).

Reply:
Regarding default hub: the code actually works if user omits hub name in the hub definition. Anyway, I prepared a separate PR to discuss this issue there. In my opinion, it's still better to make this parameter mandatory. However, it is a **breaking change**.

In cover I will enforce the `hub` parameter as required.

Reference: home-assistant/core#37546